app/src/main/java/com/example/demo/kotlin_practice/functions/Functions.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.demo.kotlin_practice.functions
+
+//infix function
+class Demo {
+    infix fun display(num: Int): Int {
+        return num
+    }
+}
+
+//recursion factorial
+fun fact(num: Int): Int {
+    if (num == 1)
+        return num
+    else
+        return num * fact(num - 1)
+}
+
+//lambdas
+val sum1 = { a: Int, b: Int -> a + b }
+val sum2: (Int, Int) -> Int = { a, b -> a + b }
+
+//shorthand lambda
+val arrNum = arrayOf(-2, 4, -5)
+
+//anonymous functions
+val an = fun(x: Int, y: Int): Int = x + y
```

Comment:
use full names, never abbreviate names, 

you are leaving chances for guesswork for the reader